idelib/transforms.py

```diff
@@ -944,14 +944,15 @@ def _build(self):
                 if p is not None:
                     for attr in ('_str', '_source', '_function', '_noY'):
                         setattr(self, attr, getattr(p, attr, None))
-        phead, src = self.poly.source.split(": ")
-            
+        phead, _, src = self.poly.source.partition(": ")
+
         for k, v in self.subpolys.items():
             if v is None:
                 ssrc = "lambda x: x"
             else:
-                ssrc = v.source 
-            s = "(%s)" % ssrc.split(": ")[-1]
+                ssrc = v.source
+            start, _, end = ssrc.rpartition(": ")
+            s = "(%s)" % (end or start)
```

Comment:
so `end` should never really just be an empty string, right? because it seems weird to fill it with `start` if not, but idk it probably doesn't make much difference 🤷

Reply:
This duplicates the behavior of the previous code if the delimiter is not in the string, but is safe against an empty string.